src/components/ICA/__tests__/ICA.spec.js

```diff
@@ -3,76 +3,75 @@
+      container: document.body.appendChild(main),
+    });
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('ICA');
   });
 
-  it('renders', () => {
-    expect(ica).toMatchSnapshot();
+  test('should render en dash when `value` is `null`', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={null} />);
+    expect(queryByText('–')).toBeVisible();
   });
 
-  it('renders the null value', () => {
-    ica.setProps({ value: null });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a large value', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={1000000} />);
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('renders a large value', () => {
-    ica.setProps({ value: largeValue });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a subset of values when `value` and `total` are provided', () => {
+    const { getByText } = render(<ICA label="test ICA" value={5} total={10} />);
+    expect(getByText('5').nextSibling).toBe(getByText(/10/i));
   });
 
-  it('renders a subset of values', () => {
-    ica.setProps({ total });
-
-    expect(ica).toMatchSnapshot();
+  test('should not render subset of values when `total` is close to `value`', () => {
+    const { queryByText } = render(
+      <ICA label="test ICA" value={1000000} total={999995} />
+    );
+    expect(queryByText('999995')).not.toBeInTheDocument();
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('does not render a subset when the total is close to the provided value', () => {
-    ica.setProps({ total: largeValue - value, value: largeValue });
-    expect(ica).toMatchSnapshot();
+  test('should not render `total` if it is the same as `value`', () => {
+    const { getByText } = render(
+      <ICA label="test ICA" value={10} total={10} />
+    );
+    // Expect 2 ICA nodes: the label and the value.
+    expect(getByText(/test ICA/i).parentNode.children.length).toBe(2);
   });
 
-  it('renders with a value that is same as the total value', () => {
-    ica.setProps({ total: value });
-
-    expect(ica).toMatchSnapshot();
-  });
-
-  it('renders a percentage', () => {
-    ica.setProps({ percentage: true, total });
-
-    expect(ica).toMatchSnapshot();
+  test('should render `total` when forced', () => {
+    const { getByText } = render(
+      <ICA label="test ICA" value={10} total={10} forceShowTotal />
+    );
+    // Expect 3 ICA nodes: the label, value, and the total.
+    expect(getByText(/test ICA/i).parentNode.children.length).toBe(3);
   });
 
-  it('renders a total when forced', () => {
-    ica.setProps({ forceShowTotal: true, total: value });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a percentage', () => {
+    const { queryByText } = render(
+      <ICA label="test ICA" value={10} percentage />
+    );
+    expect(queryByText('10').firstElementChild).toBe(queryByText('%'));
```

Comment:
Can we be consistent in our usage of `getByText` and `queryByText`? I don't mind either way as long as it's the same 🚀

Reply:
Hey there, I am consistent in my usage (`queryBy*` is to check for the existence of, because it may not return something). This appears to be an error and should be `getBy*`

Reply:
Well correction -- with edits, it will now be (so `queryBy*` is appropriate):

```
expect(queryByText('%')).toBeVisible();
```

Reply:
As @jendowns said, `getBy*` selectors will fail the test if nothing is found. In tests where you are 100% sure that the element should be there, `getBy*` is preferred. On the other hand, when you need to test if something not rendered, `queryBy*` is ideal because the tests will not fail when the selector returns `null`.

Reply:
☝️ MUCH better explanation! 😆 Thank you for those details!